src/api/goods.ts

```
@@ -4,13 +4,47 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`HTTP error! status:  ${response.status}`);
+      }
+
+      return response.json();
+    })
+    .catch(() => {
+      return [];
+    });
 }
 
 export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+  return getAll()
+    .then(goods => {
+      goods.sort((a, b) => a.name.localeCompare(b.name));
+
+      if (goods.length < 5) {
+        throw new Error('Масив меньше 5');
+      }
+
+      return goods.slice(0, 5);
+    })
+    .catch(() => {
+      return [];
+    });
 };
 
 export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+  return getAll()
+    .then(goods => {
+      const filtered = goods.filter(good => good.color === 'red');
+
+      if (filtered.length === 0) {
+        throw new Error('Немає червоних');
+      }
+
+      return filtered;
+    })
+    .catch(() => {
+      return [];
+    });
```

This violates checklist item: 'Must handle possible errors when making requests to the server (e.g., display an error state or message, avoid unhandled promise rejections)'.

Catching all errors and returning an empty array prevents the UI from knowing if an error occurred. Instead, let errors propagate so the UI can display a meaningful error message. Remove the `.catch` blocks or rethrow the error after logging.